### DISCUSSION TITLE
How can I set the wear of an item? (click for more info)

### DISCUSSION THREAD

**Discussion by aoqia194** [danielkrupinski/Osiris#3342] [Q&A] [answered]
I am trying to set the seed of an item. Reading the variable works fine, but writing doesn't.

```cpp
auto item = InventoryItem; // assume this is set correctly
auto seed = Inventory::dynamicSkinData(item.getDynamicDataIndex()).seed;
seed = 1337; // this does not work
std::cout << seed << std::endl;
```

**Comment by KurastikO** [comment, marked as answer]
I assume the seed you have inputted is >1000? If so then it'll not work as CS:GO does not recognize these types of numbers.

**Reply by aoqia194** [reply]
This doesn't make a difference.
It is physically unable to write to the variable. Even outside of CSGO.
Reading works fine.

**Reply by mdilai** [reply]
> This doesn't make a difference.
> 
> It is physically unable to write to the variable. Even outside of CSGO.
> 
> Reading works fine.

Did you tried to unprotect it first like?

https://github.com/danielkrupinski/Osiris/blob/374cd38a3e367bcfd932681ae84b5ea7b169744b/Osiris/Hooks.cpp#L604-L615

**Reply by aoqia194** [reply]
> 
> 
> > This doesn't make a difference.
> > It is physically unable to write to the variable. Even outside of CSGO.
> > Reading works fine.
> 
> Did you tried to unprotect it first like?
> 
> https://github.com/danielkrupinski/Osiris/blob/374cd38a3e367bcfd932681ae84b5ea7b169744b/Osiris/Hooks.cpp#L604-L615

That'll do the trick! Didn't even think of that...
Thank you!

**Comment by linuxer9** [comment]
seed must be less than 1000

**Comment by aoqia194** [comment]
This doesn't fix my problem.
